ricecooker/utils/youtube_cache.py

```diff
@@ -0,0 +1,167 @@
+                except Exception as e:
+                    LOGGER.error("==> [Video %s] Failed to get video info: %s", self.cachename, e)
+                    return None
+            else:
+                return None
+        return vinfo
+
+
+class YouTubePlaylistCache(object):
+
+    def __init__(self, playlist_id, alias='', cache_dir=''):
+        """
+        Initializes YouTubePlaylistCache object
+        :param alias: Alias name for the JSON cache filename, which will be named as playlist_id if such field not specified
+        """
+        self.playlist_id = playlist_id
+        if not alias:
+            self.cachename = self.video_id
+        else:
+            self.cachename = alias
+        if not cache_dir:
+            self.cache_dir = DEFAULT_YOUTUBE_CACHE_DIR
+        else:
+            self.cache_dir = cache_dir
+        if not os.path.isdir(self.cache_dir):
+            os.mkdir(self.cache_dir)
+        self.playlist_info_json_path = os.path.join(self.cache_dir, self.cachename + '.json')
+
+    def __str__(self):
+        return 'YouTubePlaylistCache (%s)' % (self.cachename)
+
+    def get_playlist_info(self, use_cache=True, youtube_skip_download=True, options=None):
+        """
+        Get YouTube playlist info by either requesting URL or extracting local cache
+        :param use_cache: Define if allowed to get playlist info from local JSON cache, default to True
+        :param youtube_skip_download: Skip the actual download of the YouTube video files
+        :param options: Additional options for youtube_dl.
+                        For full list of available options: https://github.com/ytdl-org/youtube-dl/blob/master/youtube_dl/YoutubeDL.py
+        :return: A ricecooker-like info dict info about the playlist or None if extraction fails
+        """
+        playlist_info = None
+        if os.path.exists(self.playlist_info_json_path) and use_cache:
+            LOGGER.info("==> [Playlist %s] Retrieving cached playlist information...", self.cachename)
+            playlist_info = json.load(open(self.playlist_info_json_path))
+
+        if not playlist_info:
+            playlist_url = YOUTUBE_PLAYLIST_URL_FORMAT.format(self.playlist_id)
+            playlist_resource = YouTubeResource(playlist_url)
+            youtube_extract_options = dict(
+                skip_download=youtube_skip_download,
+                extract_flat=True
+            )
+            if options:
+                youtube_extract_options.update(options)
+            if playlist_resource:
+                try:
+                    playlist_info = playlist_resource.get_resource_info(youtube_extract_options)
+                    # Traverse through the video list to remove duplicates
+                    video_set = set()
+                    videos = playlist_info.get('children')
+                    for video in videos:
+                        if video['id'] in video_set:
+                            videos.remove(video)
```

Comment:
Although I don't think there's any issue with this code specifically, generally, it's not a good idea to alter a list while you're iterating through it. Instead, I would suggest you change this to `if not video['id'] in video_set:` and add the video only in that case, and just do nothing if it's a video already in the set.